pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go

```diff
@@ -174,13 +180,56 @@ func createClusterClassBasedClusterTest(tkgctlClient tkgctl.TKGClient, deleteClu
 // createLegacyClusterTest creates and deletes (if created successfully) workload cluster
 func createLegacyClusterTest(tkgctlClient tkgctl.TKGClient, deleteClusterOptions tkgctl.DeleteClustersOptions, cliFlag bool, clusterName, namespace string) {
 	if isTKCAPIFeatureActivated {
+		var (
+			mngClient        client.Client
+			clusterResources []clusterResource
+			ctx              context.Context
+		)
 		By(fmt.Sprintf("creating TKC workload cluster, TKC-API feature-gate is activated and cli feature flag set %v", cliFlag))
 		By(fmt.Sprintf("creating TKC workload cluster %v in namespace: %v, cli feature flag is %v", clusterName, namespace, cliFlag))
 		err = tkgctlClient.CreateCluster(clusterOptions)
 		Expect(err).To(BeNil())
+		if isClusterClassFeatureActivated {
+			By(fmt.Sprintf("Get k8s client for management cluster %q", e2eConfig.ManagementClusterName))
+			mngkubeConfigFileName := e2eConfig.ManagementClusterName + ".kubeconfig"
+			mngtempFilePath := filepath.Join(os.TempDir(), mngkubeConfigFileName)
+			err = tkgctlClient.GetCredentials(tkgctl.GetWorkloadClusterCredentialsOptions{
```

Comment:
We dont need to do this. The kubeconfig file is available as part of e2eConfig.TKGSKubeconfigPath and e2eConfig.TKGSKubecontext.